test/server/transformation-controller.spec.js

```diff
@@ -0,0 +1,258 @@
+const expect = require( 'chai' ).expect;
+const request = require( 'supertest' );
+const sinon = require( 'sinon' );
+const communicator = require( '../../app/lib/communicator' );
+const accountModel = require( '../../app/models/account-model' );
+const config = require( '../../app/models/config-model' ).server;
+const surveyModel = require( '../../app/models/survey-model' );
+const userModel = require( '../../app/models/user-model' );
+
+/**
+ * @typedef {import('../../app/models/survey-model').SurveyObject} Survey
+ */
+
+describe( 'Transformation Controller', () => {
+    const basePath = '';
+    const bearer = 'fozzie';
+    const credentials = { bearer };
+    const enketoId = 'surveyZ';
+    const manifestHost = 'http://example.com';
+    const manifestPath = '/manifest';
+    const manifestUrl = `${manifestHost}${manifestPath}`;
+
+    /** @type {import('sinon').SinonSandbox} */
+    let sandbox;
+
+    /** @type {import('express').Application} */
+    let app;
+
+    /** @type {import('http').Server} */
+    let server;
+
+    /** @type {import('../../app/models/account-model').AccountObj */
+    let account;
+
+    /** @type {Survey} */
+    let survey;
+
+    /** @type {Survey} */
+    let surveyWithAccount;
+
+    beforeEach( done => {
+        sandbox = sinon.createSandbox();
+
+        sandbox.stub( config, 'base path' ).get( () => basePath );
+
+        // Stub `_getSurveyParams`
+        survey = {
+            info: {
+                manifestUrl,
+            },
+        };
+
+        sandbox.stub( surveyModel, 'get' ).resolves( survey );
+
+        account = {};
+
+        surveyWithAccount = Object.assign( {}, survey, { account } );
+
+        sandbox.stub( accountModel, 'check' ).resolves( surveyWithAccount );
+
+        // No-op `_checkQuota`
+        sandbox.stub( config, 'account lib' ).get( () => null );
+
+        sandbox.stub( userModel, 'getCredentials' ).resolves( credentials );
+
+        app = require( '../../config/express' );
+
+        server = app.listen( () => done() );
+    } );
+
+    afterEach( done => {
+        sandbox.restore();
+        server.close( done );
+    } );
+
+    const requests = [
+        {
+            description: 'new survey',
+            url: '/transform/xform',
+            body: { xformUrl: 'http://example.com/qwerty' },
+        },
+        {
+            description: 'existing survey',
+            url: `/transform/xform/${enketoId}`,
+        },
+    ];
+
+    requests.forEach( ( { description, url, body } ) => {
+        describe( `jr: media URLs ${description}`, () => {
+            /** @type {import('enketo-transformer/src/transformer').TransformedSurvey} */
+            let result;
+
+            beforeEach( done => {
+                // Stub getXForm
+                const xform = `
+                    <?xml version="1.0"?>
```

Comment:
There's precedent for external fixture files in core/transformer. In my experience it makes it harder to reason about the tests using them, because you need at least three files open to understand the tested behavior, the state of the test, and the test itself.

I have similar concern about moving it below, but more in the sense that it feels odd to me to have references to something before its definition, and tends to obscure context.

Is the concern that it's large and pushes the actual tests down? I know that may be annoying for this review, but FWIW most editors/IDEs will support collapsing it so it shouldn't be too big a burden going forward.